File: mahaam-api-cs/Src/Infra/Cache.cs
using System.Net;
using System.Net.Sockets;
using Mahaam.Infra.Monitoring;


namespace Mahaam.Infra;

public interface ICache
{
	void Init(Health health);
	string NodeIP();
	string NodeName();
	Guid HealthId();
}

class Cache : ICache
{
	private Health? _health;

	public void Init(Health health)
	{
		_health = health;
	}

	public string NodeIP() => _health?.NodeIP ?? "";
	public string NodeName() => _health?.NodeName ?? "";
	public Guid HealthId() => _health?.Id ?? Guid.Empty;
}

File: mahaam-api-go/infra/cache/cache.go
package cache

import (
	"mahaam-api/feat/models"

	"github.com/google/uuid"
)

var (
	NodeIP     string
	NodeName   string
	ApiName    string
	ApiVersion string
	EnvName    string
	HealthID   uuid.UUID
)

func Init(h *models.Health) {
	NodeIP = h.NodeIP
	NodeName = h.NodeName
	ApiName = h.ApiName
	ApiVersion = h.ApiVersion
	EnvName = h.EnvName
	HealthID = h.ID
}
